ui/app/components/transit-key-actions.js

```diff
@@ -35,213 +50,142 @@ const TRANSIT_PARAMS = {
   exportKeyType: null,
   exportKeyVersion: null,
   wrappedToken: null,
+  wrappedTTL: '30m',
   valid: null,
   plaintextOriginal: null,
   didDecode: false,
   verification: 'Signature',
 };
-const PARAMS_FOR_ACTION = {
+
+const SUCCESS_MESSAGE_FOR_ACTION = {
+  sign: 'Signed your data.',
+  // the verify action doesn't trigger a success message
+  hmac: 'Created your hash output.',
+  encrypt: 'Created a wrapped token for your data.',
+  decrypt: 'Decrypted the data from your token.',
+  rewrap: 'Created a new token for your data.',
+  datakey: 'Generated your key.',
+  export: 'Exported your key.',
+};
+
+const PROPS_TO_KEEP = {
+  encrypt: ['plaintext', 'context', 'nonce', 'key_version'],
+  decrypt: ['ciphertext', 'context', 'nonce'],
   sign: ['input', 'hash_algorithm', 'key_version', 'prehashed', 'signature_algorithm'],
   verify: ['input', 'hmac', 'signature', 'hash_algorithm', 'prehashed'],
   hmac: ['input', 'algorithm', 'key_version'],
-  encrypt: ['plaintext', 'context', 'nonce', 'key_version'],
-  decrypt: ['ciphertext', 'context', 'nonce'],
   rewrap: ['ciphertext', 'context', 'nonce', 'key_version'],
+  datakey: [],
```

Comment:
in the old code we used a logic statement to catch the "datakey" case. I opted to return an empty array here instead to help clean up the logic farther down.

Reply:
Before these were abstracted because the `component` helper was being used - could we pass these args directly into the template now?

Reply:
I think this would clean up the component logic quite a bit. Normally I wouldn't suggest if this PR was strictly glimmerizing but since you did the additional refactoring in the template, it seems reasonable to clean this up as well